components/script/dom/xmlhttprequest.rs

```diff
@@ -584,14 +584,22 @@ impl XMLHttpRequestMethods for XMLHttpRequest {
                     source: BodySource::Null,
                 })
             },
-            Some(DocumentOrBodyInit::Blob(ref b)) => Some(b.extract(&self.global())),
-            Some(DocumentOrBodyInit::FormData(ref formdata)) => {
-                Some(formdata.extract(&self.global()))
+            Some(DocumentOrBodyInit::Blob(ref b)) => {
+                Some(b.extract(&self.global()).expect("Couldn't extract body."))
             },
-            Some(DocumentOrBodyInit::String(ref str)) => Some(str.extract(&self.global())),
-            Some(DocumentOrBodyInit::URLSearchParams(ref urlsp)) => {
-                Some(urlsp.extract(&self.global()))
+            Some(DocumentOrBodyInit::FormData(ref formdata)) => Some(
+                formdata
+                    .extract(&self.global())
+                    .expect("Couldn't extract body."),
+            ),
+            Some(DocumentOrBodyInit::String(ref str)) => {
+                Some(str.extract(&self.global()).expect("Couldn't extract body."))
             },
+            Some(DocumentOrBodyInit::URLSearchParams(ref urlsp)) => Some(
+                urlsp
+                    .extract(&self.global())
+                    .expect("Couldn't extract body."),
```

Comment:
Shouldn't we handle the error instead of panicking?

Reply:
We could, although it's a similar situation to the `Vec<u8>` where the extraction cannot actually fail.